src/realm/object-store/c_api/sync.cpp

```diff
@@ -413,6 +413,24 @@ RLM_API void realm_sync_config_set_resync_mode(realm_sync_config_t* config,
     config->client_resync_mode = ClientResyncMode(mode);
 }
 
+RLM_API realm_flx_sync_subscription_set_t* get_latest_subscription_set(const realm_t* realm)
+{
+    if (realm == nullptr)
+        return nullptr;
+
+    auto subscription = (*realm)->get_latest_subscription_set();
+    return new realm_flx_sync_subscription_set_t(std::make_shared<sync::SubscriptionSet>(subscription));
+}
+
+RLM_API realm_flx_sync_subscription_set_t* get_active_subscription_set(const realm_t* realm)
+{
+    if (realm == nullptr)
+        return nullptr;
+
+    auto subscription = (*realm)->get_active_subscription_set();
+    return new realm_flx_sync_subscription_set_t(std::make_shared<sync::SubscriptionSet>(subscription));
```

Comment:
I am not sure if I can just return a pointer to the internal subscription or, as I am doing in here, copy construct a new subscription set and return a shared_ptr towards the sdks. @jedelbo  what's the correct approach here? ... I noticed we tend to construct a shared_ptr and pass that to the sdks most of the times.

Reply:
I think we should return
```
new realm_flx_sync_subscription_set_t((*realm)->get_active_subscription_set());
```